sources/packages/backend/apps/queue-consumers/src/processors/schedulers/student-application-notifications/student-application-notifications.scheduler.ts

```
@@ -0,0 +1,62 @@
+import { InjectQueue, Process, Processor } from "@nestjs/bull";
+import { QueueService } from "@sims/services/queue";
+import { Job, Queue } from "bull";
+import { BaseScheduler } from "../base-scheduler";
+import {
+  InjectLogger,
+  LoggerService,
+  ProcessSummary,
+} from "@sims/utilities/logger";
+import {
+  getSuccessMessageWithAttentionCheck,
+  logProcessSummaryToJobLogger,
+} from "../../../utilities";
+import { QueueNames } from "@sims/utilities";
+import { ApplicationService } from "../../../services";
+import { NotificationService } from "@sims/services/notifications";
+
+@Processor(QueueNames.StudentApplicationNotifications)
+export class StudentApplicationNotificationsScheduler extends BaseScheduler<void> {
+  constructor(
+    @InjectQueue(QueueNames.StudentApplicationNotifications)
+    schedulerQueue: Queue<void>,
+    queueService: QueueService,
+    private readonly applicationService: ApplicationService,
+    private readonly notificationService: NotificationService,
+  ) {
+    super(schedulerQueue, queueService);
+  }
+
+  @Process()
+  async StudentApplicationNotifications(job: Job<void>): Promise<string[]> {
+    const processSummary = new ProcessSummary();
+
+    try {
+      this.logger.log(
+        `Processing student application notifications job. Job id: ${job.id} and Job name: ${job.name}.`,
+      );
+
+      const eligibleApplications =
+        await this.applicationService.getEligibleApplicationsForNotification();
+      // TODO: Get applications that have a disability status mismatch  and check PDPPD status
+
+      processSummary.info(` ${eligibleApplications}`);
+
+      return getSuccessMessageWithAttentionCheck(
+        ["Process finalized with success."],
+        processSummary,
+      );
+    } catch (error: unknown) {
+      const errorMessage = "Unexpected error while executing the job.";
+      processSummary.error(errorMessage, error);
+      return [errorMessage];
```

Please throw the error here to allow the job to retry.
```ts
throw new Error(errorMessage, { cause: error });
```